src/Bicep.Core/Analyzers/Linter/LinterAnalyzer.cs

```diff
@@ -50,7 +47,13 @@ public LinterAnalyzer(RootConfiguration configuration)
 
             foreach (var ruleType in ruleTypes)
             {
-                rules.Add(Activator.CreateInstance(ruleType) as IBicepAnalyzerRule ?? throw new InvalidOperationException($"Failed to create an instance of \"{ruleType.Name}\"."));
+                if (Activator.CreateInstance(ruleType) is IBicepAnalyzerRule rule)
+                {
+                    rules.Add(rule);
+                } else 
+                {
+                    errors.Add(DiagnosticBuilder.ForDocumentStart().RuleFailedToLoad(ruleType.Name));
+                }
```

Comment:
Looks like this may have been based off an old copy of the `main` branch? Stephen deliberately removed the conversion of exception -> diagnostic under #8194

Reply:
I misinterpreted what you said in https://github.com/Azure/bicep/pull/8301#discussion_r966212202. Reverted to old behavior.